### PR TITLE
doc: nrf5340audio: clean up headset and gateway on architecture page

### PR DESCRIPTION
**Pull request by greg-fer**
Cleaned up gateway and headset references.
OCT-2954.

### CODE REVIEW

File: applications/nrf5340_audio/doc/firmware_architecture.rst [hunk 1]
```diff
@@ -7,14 +7,28 @@ nRF5340 Audio overview and firmware architecture
...
+
+  * Unicast client
+  * Unicast server
+
+* For the *broadcast isochronous stream* (BIS) mode:
+
+  * Broadcast source
+  * Broadcast sink
+
+Unicast client (CIS) and broadcast source (BIS) receive the audio data from external sources (USB or line input/I2S) and forward it to one or more unicast servers or broadcast sinks, thus becoming source devices for them.
```
Comment (koffes) [inline applications/nrf5340_audio/doc/firmware_architecture.rst:22]:
```suggestion
Unicast client (CIS) and broadcast source (BIS) receive the audio data from external sources (USB or line input/I2S) and forward it to one or more unicast servers or broadcast sinks respectively.
```

File: applications/nrf5340_audio/doc/firmware_architecture.rst [hunk 2]
```diff
@@ -7,14 +7,28 @@ nRF5340 Audio overview and firmware architecture
...
+Unicast servers (CIS) and broadcast sinks (BIS) are receiver devices that play back the audio they get from source devices, and will act as earbuds, headphones, speakers, hearing aids, or similar.
```
Comment (koffes) [inline applications/nrf5340_audio/doc/firmware_architecture.rst:23]: This is where we may rephrase a bit. A broadcast sink is always a receiver, but a unicast server can be a sink and a source.

File: applications/nrf5340_audio/doc/firmware_architecture.rst [hunk 3]
```diff
@@ -7,14 +7,28 @@ nRF5340 Audio overview and firmware architecture
...
+Unicast servers can also send audio back to the unicast client.
 See :ref:`nrf53_audio_app_overview_modes` for more information.
 
-The applications use the same code base, but use different :file:`main.c` files and include different modules and libraries depending on the configuration.
+The nRF5340 Audio applications use the same code base, but use different :file:`main.c` files and include different modules and libraries depending on the configuration.
+
+.. important::
+    Before the |NCS| v2.6.0, there was a single nRF53 Audio application with different configuration options for gateway (currently, unicast client and broadcast source) and headset (currently, unicast server and broadcast sink).
```
Comment (koffes) [inline applications/nrf5340_audio/doc/firmware_architecture.rst:30]:
```suggestion
    Before the |NCS| v2.6.0, there was a single nRF53 Audio application with different configuration options for gateway ( unicast client and broadcast source) and headset ( unicast server and broadcast sink).
```

File: applications/nrf5340_audio/doc/firmware_architecture.rst [hunk 4]
```diff
@@ -7,14 +7,28 @@ nRF5340 Audio overview and firmware architecture
...
+    For this reason, until it is cleaned up, the code and documentation of the nRF5340 Audio applications still make references to gateway and headset.
```
Comment (koffes) [inline applications/nrf5340_audio/doc/firmware_architecture.rst:31]: Remove this sentence. We may keep these definitions for quite a while longer.

File: applications/nrf5340_audio/doc/firmware_architecture.rst [hunk 5]
```diff
@@ -32,31 +46,30 @@ Each application works either in the *connected isochronous stream* (CIS) mode o
...
-  CIS is a bidirectional communication protocol that allows for sending separate connected audio streams from a source device to one or more receivers.
-  The gateway can send the audio data using both the left and the right ISO channels at the same time, allowing for stereophonic sound reproduction with synchronized playback.
+  CIS is a bidirectional (connected) communication protocol that allows for sending separate connected audio streams from a source device to one or more receivers.
+  The source can send the audio data using both the left and the right ISO channels at the same time, allowing for stereophonic sound reproduction with synchronized playback.
 
-  This is the mode available for the unicast applications (:ref:`unicast client<nrf53_audio_unicast_client_app>` and :ref:`unicast server<nrf53_audio_unicast_server_app>`).
-  In this mode, you can use the nRF5340 Audio development kit in the role of the gateway, the left headset, or the right headset.
+  This is the mode available for the unicast applications (:ref:`unicast client<nrf53_audio_unicast_client_app>` as source and :ref:`unicast server<nrf53_audio_unicast_server_app>` as receivers).
```
Comment (koffes) [inline applications/nrf5340_audio/doc/firmware_architecture.rst:52]:
```suggestion
  This is the mode available for the unicast applications (:ref:`unicast client<nrf53_audio_unicast_client_app>` as gateway and :ref:`unicast server<nrf53_audio_unicast_server_app>` as headset).
```

File: applications/nrf5340_audio/doc/firmware_architecture.rst [hunk 6]
```diff
@@ -32,31 +46,30 @@ Each application works either in the *connected isochronous stream* (CIS) mode o
...
 
   In the current version of the nRF5340 Audio unicast client, the application offers both unidirectional and bidirectional communication.
-  In the bidirectional communication, the headset device will send audio from the on-board PDM microphone.
+  In the bidirectional communication, the receiver device will send audio from the on-board PDM microphone.
```
Comment (koffes) [inline applications/nrf5340_audio/doc/firmware_architecture.rst:55]:
```suggestion
  In the bidirectional communication, the headset device will send audio from the on-board PDM microphone.
```

File: applications/nrf5340_audio/doc/firmware_architecture.rst [hunk 7]
```diff
@@ -32,31 +46,30 @@ Each application works either in the *connected isochronous stream* (CIS) mode o
...
   See :ref:`nrf53_audio_app_configuration_select_bidirectional` in the application description for more information.
 
   You can also enable a walkie-talkie demonstration.
-  In this demonstration, the gateway device will send audio from the on-board PDM microphone instead of using USB or the line-in.
+  In this demonstration, the source device will send audio from the on-board PDM microphone instead of using USB or the line-in.
```
Comment (koffes) [inline applications/nrf5340_audio/doc/firmware_architecture.rst:59]: In this config both devices will be source and sink.

File: applications/nrf5340_audio/doc/firmware_architecture.rst [hunk 8]
```diff
@@ -32,31 +46,30 @@ Each application works either in the *connected isochronous stream* (CIS) mode o
...
 
 Broadcast Isochronous Stream (BIS)
-  BIS is a unidirectional communication protocol that allows for broadcasting one or more audio streams from a source device to an unlimited number of receivers that are not connected to the source.
+  BIS is a unidirectional (broadcast/disconnected) communication protocol that allows for broadcasting one or more audio streams from a source device to an unlimited number of receivers that are not connected to the source.
 
-  This is the mode available for the broadcast applications (:ref:`broadcast source<nrf53_audio_broadcast_source_app>` for headset and :ref:`broadcast sink<nrf53_audio_broadcast_sink_app>` for gateway).
-  In this mode, you can use the nRF5340 Audio development kit in the role of the gateway or as one of the headsets.
-  Use multiple nRF5340 Audio development kits to test BIS having multiple receiving headsets.
+  This is the mode available for the broadcast applications (:ref:`broadcast source<nrf53_audio_broadcast_source_app>` for receivers and :ref:`broadcast sink<nrf53_audio_broadcast_sink_app>` for source devices).
```
Comment (koffes) [inline applications/nrf5340_audio/doc/firmware_architecture.rst:65]:
```suggestion
  This is the mode available for the broadcast applications (:ref:`broadcast source<nrf53_audio_broadcast_source_app>` for gateways and :ref:`broadcast sink<nrf53_audio_broadcast_sink_app>` for headset devices).
```

File: applications/nrf5340_audio/doc/firmware_architecture.rst (hunk 2 above)
Comment (greg-fer) [reply applications/nrf5340_audio/doc/firmware_architecture.rst:23]: I think this is clarified in line 24?

**Comment by greg-fer** [comment]
Closing based on comment from @koffes on Teams: "we have not yet agreed to remove the "gateway" and "headset" definitions. We should remove those from the doc at the same time as we remove them from the code".

**Pull request closed without merging**